### ISSUE TITLE
Add ignore: ["child"] option to selector-no-type

### ISSUE THREAD

**Issue by mik01aj** [stylelint/stylelint#1942]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Request for enhancement in `selector-no-type`

> Which rule, if any, is this issue related to?

[`selector-no-type`](https://github.com/stylelint/stylelint/tree/master/src/rules/selector-no-type) 

> What CSS is needed to reproduce this issue?

```
.className > label {} // I want to allow this...
.className label {}   // ...but not this
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

``` json
{
  "rules": {
        "selector-no-type": [true, {
            ignore: ["compounded", "descendant"],
        }],
  }
}
```

> Which version of stylelint are you using?

`7.3.1`.

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

Hmm. With that config, perhaps just the 1st one should warn, because it's a child selector, not descendant. It would be good to control descendants and children individually. Then I could allow the 1st but flag the 2nd line.

> What actually happened (e.g. what warnings or errors you are getting)?

With this config, both lines are fine. When I remove `descendant`, both are flagged as errors.

**Comment by jeddy3** [comment]
@mik01aj Thank you for using the issue template. Having the question and answer format helps us parse through these things more easily.

@stylelint/core This seems like a valid request to me i.e the addition of a `ignore: ["child"]` optional secondary option. Does anyone have any objections to this? If not, I'll label up the issue accordingly and point @mik01aj in the direction of how to add the option.

**Comment by davidtheclark** [comment]
👍

**Comment by jeddy3** [comment]
@mik01aj Would you like to contribute this option? There's a [specific section](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules) in the developer guide to help people contribute new options. You'll want to check if the *type selector* is preceded by *child combinator*. 
 
> Hmm. With that config, perhaps just the 1st one should warn, because it's a child selector, not descendant.

Good spot. I'll create a separate issue for this. Lets just focus on adding the `ignore: ["child']` option in this issue though, as I think it'll address your particular use-case.

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/2217
